enhancements/cluster-logging/gcp-external-account-auth.md

```diff
@@ -0,0 +1,201 @@
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement adds support for Workload Identity Federation (WIF) to the existing GCP 
+authentication mechanism in Vector. With this update, Vector will be able to authenticate 
+using external_account credentials when running in an OpenShift environment, enabling OIDC-based 
+authentication with short-lived access tokens instead of static credentials.
+
+## Motivation
+
+Current GCP authentication in Vector relies on service account credentials, which require 
+static key management. WIF eliminates the need for long-lived credentials, enabling more secure, 
+short-lived authentication via OpenShift service account tokens.
+
+### User Stories
+
+* As an OpenShift administrator, I want Vector to authenticate with GCP using Workload Identity Federation, 
+removing the need for static service account keys. 
+* As an engineer, I want Vector to detect the correct authentication method based on the credentials file 
+format, utilizing the correct credentials chain form the sdk, and ensuring compatibility with 
+both service accounts and external accounts (WIF).
+
+
+### Goals
+
+* Modify the GCP module in vector to support external_account credentials. 
+* Introduce new structs (ExternalCredentials, CredentialsType) to handle the updated credential format. 
+* Serialize the configuration file into usable data 
+* Implement the full OIDC-based authentication flow. 
+* Ensure compatibility with existing service account authentication
+
+
+### Non-Goals
+
+* Modifying the OpenShift gcp output authentication model. 
+* Supporting authentication methods outside of Google Cloud.
+
+
+## Proposal
+
+To integrate WIF, Vector’s authentication logic will be updated to support external_account 
+credentials in addition to the existing service_account credentials. The process will:
+
+* Determine the authentication type by reading the credentials JSON file. 
+* Extract the OpenShift Service Account token from the projected volume. 
+* Exchange the OpenShift token for a Google Identity Token using Google STS. 
+* Exchange the Identity Token for a short-lived Access Token via Google IAM impersonation. 
+* Ensure the existing service account authentication remains functional.
+
+### Workflow Description
+### API Extensions
+The existing credentials key and secret works for our purpose, although we should aim to eventually  
+align with our cloudwatch and future Azure specs which are most likely going to be configMaps:
+```yaml
+ googleCloudLogging:
+   authentication:
+     credentials:
+       key: google-ext-account-creds.json
+       secretName: gcp-secret
+   logId : my-gcp
+...
+```
+### Topology Considerations
+#### Hypershift / Hosted Control Planes
```

Comment:
Is there any need to mention who rotates the token or how this is handled in HCP?  In our conversations maybe you mentioned this is abstracted to us by the OIDC provider?

Reply:
The openshift projected token is considered a "bound service account token" and the kubelet rotates the token if it is older than 80 percent of its time to live (or 24 hours, whichever is less).    The application that uses the bound token must only handle reloading the token after it rotates.  This is handled by the vector client regardless of where it gets the token.

Reply:
I'm only asking the question for completeness and defer to you if there is anything additional to mention.

Reply:
original content has been removed